pkg/serving/config_changes.go

```diff
@@ -331,14 +331,32 @@ func UpdateContainerArg(template *servingv1.RevisionTemplateSpec, arg []string)
 	return nil
 }
 
-// UpdateContainerPort updates container with a give port
-func UpdateContainerPort(template *servingv1.RevisionTemplateSpec, port int32) error {
+// UpdateContainerPort updates container with a given name:port
+func UpdateContainerPort(template *servingv1.RevisionTemplateSpec, port string) error {
 	container, err := ContainerOfRevisionTemplate(template)
 	if err != nil {
 		return err
 	}
+	var containerPort int64
+	var name string
+
+	if strings.Contains(port, ":") {
+		portDetails := strings.Split(port, ":")
+		name = portDetails[0]
```

Comment:
We should check for the length of portDetails here and empty string values.
The test cases could then verify sample input as:

- `h2c:8080`  :heavy_check_mark: 
- `:8080` :heavy_check_mark: (name: `""`, port: `8080`)
- `h2c:` :x: (empty port, `ParseInt` would throw error in this case if not catched earlier, raise error hinting correct format)
- `foo:8080:proto` :x: ( raise error hinting correct format)

Reply:
have made the changes. Please take a look